WordPress/gutenberg
# Components: refactor`Dropdown` to pass `exhaustive-deps`

## What?
Updates the `Dropdown` component to satisfy the `exhaustive-deps` eslint rule

## Why?
Part of the effort in https://github.com/WordPress/gutenberg/pull/41166 to apply `exhuastive-deps` to the Components package

## How?
Adds `onToggle` to the `useEffect` dependency array. We don't expect `onToggle` to change, so this should satisfy `exhaustive-deps` while being equivalent to an empty dep array.

## Testing Instructions
1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/dropdown`
2. Confirm that the linter returns no errors
3. Confirm unit tests still pass
4. Run Storybook locally, confirm the components stories and/or docs still work as expected
